docgen/src/guides/routing.md

```diff
@@ -0,0 +1,276 @@
+
+## User friendly urls
+
+You might want your urls to look like:
+
+`https://website.com/?query=a&brands=Sony~Samsung&page=2`
+
+This way your users to be able to more easily read them when shared via emails, documents, social media...
+
+To do so, the `routing` option accepts a simple boolean but also more complex objects to allow for customization. The first customization option
+you want to use is `stateMapping`. It allows you to define more precisely how the state of your search will be synchronized to your
+browser bar (or any other router storage you might have).
+
+Here's an example achieving just that ([and here's the live version](https://codesandbox.io/s/005py264m0)):
+
+This example assumes that you have added the [searchBox](widgets/searchBox.html), [refinementList](widgets/refinementList.html)
+and [pagination](widgets/pagination.html) widgets to your search UI. Then the `refinementList` is activated on the `brands` attribute.
+Please adjust given your own data.
+
+```javascript
+const search = instantsearch({
+  // ... appId, apiKey...
+  routing: {
+    stateMapping: {
+      stateToRoute(uiState) {
+        return {
+          query: uiState.query,
+          // we use the character ~ as it is one that is rarely present in data and renders well in urls
+          brands: uiState.refinementList && uiState.refinementList.brand.join('~'),
+          page: uiState.page
+        };
+      },
+      routeToState(routeState) {
+        return {
+          query: routeState.query,
+          refinementList: {
+            brand: routeState.brands && routeState.brands.split('~')
+          },
+          page: routeState.page
+        };
+      }
+    }
+  }
+});
+```
+
+There's a lifecycle dictating when the `stateMapping` functions are called:
+- `stateToRoute` is called whenever widgets are refined (clicked). It is also called everytime any widget needs to create a URL.
+- `routeToState` is called whenever the user loads, reloads the page or click on back/next buttons of the browser.
+
+To build your own mapping easily, just `console.log(uiState)` and see what you're getting. Note that the object you return in `stateToRoute` will be the one
+you'll receive as an argument in `routeToState`.
+
+## SEO friendly urls
+
+You might also want your urls to look like:
+
+`https://website.com/search/q/phone/brands/Sony~Samsung/p/1`
+
+Be it for SEO benefits or to align your search UI urls with your current sitemap and existing url scheme.
+
+Here's an example achieving just that ([and here's the live version](https://codesandbox.io/s/l5z6zz4yoq)):
+
+```javascript
+const search = instantsearch({
+  // ... appId, apiKey...
+  routing: {
+    router: instantsearch.routers.history({
+      windowTitle: function(routeState) {
```

Comment:
we can use either all modern features (object shorthand) or none (no template strings either) so that people can copy-paste it in sites without build process

Reply:
I want to keep "routeState" here for the sake of explaining which kind of objects you are receiving. Users can then write that however they want. Works?

Reply:
I meant:

```diff
- windowTitle: function(routeState) {
+ windowTitle(routeState) {
```